### DISCUSSION TITLE
Unable to run the promotheus ansible-pull command  ERROR: ERROR! Specified inventory, host pattern and/or --limit leaves us with no hosts to target.

### DISCUSSION THREAD

**Discussion by NabiDasta** [learndevopsonline/learndevopsonline#1653] [Q&A] [answered]
If i run the ansible-pull command it's showing the below error could you please help me. checked multiple times the command is correct only. I am not getting what is the error here.

**ansible-pull -i 172.31.38.199, -U https://github.com/NabiDasta/Expense-ansible expense.yml -e role_name=prometheus**

![image](https://github.com/learndevopsonline/learndevopsonline/assets/148522896/94f903c6-841b-4b80-aedb-bce70aa5a99e)

My Public IP: 52.87.194.176
      Private IP: 172.31.38.199

**Comment by sreedharm07** [comment, marked as answer]
@NabiDasta 
in command instead of ip give as localhost it will work

**Comment by NabiDasta** [comment]
Thank you it's working. But getting one more ERROR . Every time i am getting same error can u help me this as well. Even Prometheus server also not running properly. it means I didn't configure well ??

![image](https://github.com/learndevopsonline/learndevopsonline/assets/148522896/4b3b8b47-3f8e-4dcd-bd52-231b1b6b6af9)
 
![image](https://github.com/learndevopsonline/learndevopsonline/assets/148522896/c84c5124-e51d-48e6-95b2-bd5c0811b0d2)

![image](https://github.com/learndevopsonline/learndevopsonline/assets/148522896/d8d775b7-110a-4413-9369-9d42baebf5dd)

**Reply by sreedharm07** [reply]
create alert-rules folder manually in path -- /opt/prometheus/

**Reply by NabiDasta** [reply]
Thanks your patience to respond .
Yes sreedhar if i create the path manually in server it's working I know that before only. I just want to know why it's not taking automation through ansible. where i did mistake ??

**Reply by NabiDasta** [reply]
![image](https://github.com/learndevopsonline/learndevopsonline/assets/148522896/de64e6a3-5675-4178-9524-3e54a31aa2e7)

See again I stopped instance and again start the instance. Above Error I am getting every time like failed status in Prometheus server. Even I checked the service as well but there also not installed Prometheus
 if you see **netstat -lntp**

**Reply by sreedharm07** [reply]
@NabiDasta 
there is nothing mistake, you are telling in script to copy file in below path, but there is no alert-rules folder so you should create it 
![image](https://github.com/learndevopsonline/learndevopsonline/assets/137472068/cf45bac1-0dea-48c4-a3bb-a8755249e803)

**Reply by NabiDasta** [reply]
Thank you so much. Can u give an explanation about the failed status as well Sreedhar. like before you reply i give one more error as well like failed status in Prometheus server every time if i stop and start instance in AWS.

**Reply by sreedharm07** [reply]
@NabiDasta 
just now i have runned ansible script once again can you turn off instance and try once
![image](https://github.com/learndevopsonline/learndevopsonline/assets/137472068/9bc23aef-3a3d-4ed9-80fc-b78772b05542)

**Reply by NabiDasta** [reply]
![image](https://github.com/learndevopsonline/learndevopsonline/assets/148522896/f262cbc8-d6ff-4ee6-8e60-f427a259b846)

Again same ERROR happening Sreedhar. If i again install the prometheus in labauto then it will active so every time same thing happening.

**Reply by sreedharm07** [reply]
after installing give systemctl enable prometheus 
and try once

**Reply by NabiDasta** [reply]
Yes after install and enable it will work but my concern is every time if i stop and start the system in AWS again same ERROR happening like Failed status in Prometheus. Again I re installing prometheus then only it's Active status.

![image](https://github.com/learndevopsonline/learndevopsonline/assets/148522896/15b9b9cb-3964-4c7d-af5d-64366a51cfeb)

**Reply by sreedharm07** [reply]
ya for that only 
give enable command and turn off and try 
because enable command will turn on automatically when you turn on machine

**Reply by NabiDasta** [reply]
OK Sreedhar will see If i get sam error again I will join the Lab session and will check.

But what about below Error sreedhar. Just now You run it ansible-pull prometheus right that time it's fine but just now again i run it same ERROR i am getting . It means every time we should create the path ??

![image](https://github.com/learndevopsonline/learndevopsonline/assets/148522896/beb96378-2758-4c1c-be9e-7d00abe3667e)

**Reply by sreedharm07** [reply]
in your prometheus.yml file i see some error please re write that 
thats happening the errors
instead of private_ip given private, instance instead of instance_id 
and check promethus.yml file again
![image](https://github.com/learndevopsonline/learndevopsonline/assets/137472068/ca02f1b4-c4d9-4d37-a6ad-a58710940292)

**Reply by NabiDasta** [reply]
Thank you sreedhar i will join the lab session will discuss above Errors

**Comment by NabiDasta** [comment]
Thank you sreedhar i will join the lab session will discuss above Errors
